math/dec.go

```diff
@@ -576,3 +577,26 @@ func MaxDec(x, y Dec) Dec {
 	}
 	return y
 }
+
+// MulInt multiplies the decimal by an int64 and returns the result or an error.
+func (x Dec) MulInt(y int64) (Dec, error) {
```

Comment:
May be rename function to `MulRawInt` (similar to LegacyDec) as MulInt can be confused with multiplying with `math.Int`